### ISSUE TITLE
How did you train your instagan model?

### ISSUE THREAD

**Issue by hjwang-824** [sangwoomo/instagan#15]
I trained the pix2pix model to predict the pants mask and the skirts mask, and the result of pix2pxi model were satisfied, then I trained the instagan model, but the result seems bad. So how did you train your instagan model? Have you combined the skirts and skin mask together, and train the instagan to learn to transfer pants to skirts and the skin of legs?

**Comment by sangwoomo** [comment]
Hi, did you trained InstaGAN model with predicted masks?
In my experience, the performance of InstaGAN was pretty sensitive to the mask qualities. Hence, although the results of pix2pix model is "pretty good", it may not enough for training of InstaGAN.
In my experiments, I trained InstaGAN with "real masks" and only used predicted masks for inference. Improving the robustness to the mask quality would be an interesting research direction.

**Comment by hjwang-824** [comment]
Thank you. I train the instagan with real mask, but test it with the predict masks.

**Comment by hjwang-824** [comment]
I used real masks to train the instagan model by running: python train.py --dataroot ./datasets/jeans2skirts_ccp_crop --model insta_gan --name jeans2skirts_ccp_crop_instagan --loadSizeH 200 --loadSizeW 80 --fineSizeH 200 --fineSizeW 80 --niter 400 --niter_decay 200 --batch_size 3 --gpu_ids 0,1,2. And I just test it with real mask of jeans and skirts, and the result is very bad either.

**Comment by sangwoomo** [comment]
Hi, how about performance on your training set? Can you try with the pretrained network?
Also, as the training set and test set are different, although it performs okay for training set, it may fail for test set. In my experience, front-looking images were easier to transform.

**Comment by hjwang-824** [comment]
Hi, performance on on my training set is bad either. The perfomance of pretrain network is good on the CCP dataset, but not good enough on some low pixel photos.

**Comment by sangwoomo** [comment]
As the pretrained networks are trained on high-resolution images, it'd be better to match the resolution. How about scale up your low-resolution images and use pretrained networks?

**Comment by hjwang-824** [comment]
That's a good idea. But I want to apply  the model to the low resolution photos, so I compressed the image of the CCP data set and tried to train to instaGAN model with the real mask and those low resolution photos. However the trainning result were not satisfied.

**Comment by sangwoomo** [comment]
Why not translate high-resolution images and resize it to low-resolution?
Maybe the current model parameter is too big for low-resolution images. You can try smaller models.

**Comment by hjwang-824** [comment]
OK, I will try, thank you very much! And I just find the result of the 200th epoch on the training set is pretty good and the performance of 600th epoch is poor.

**Comment by sangwoomo** [comment]
Great! Maybe 600 epoch is too much for low-resolution images (overfitting?). Early stopping could be a solution :)

**Comment by hjwang-824** [comment]
Yeah, that's what I think.
